### DISCUSSION TITLE
take long time to  exec "optimize table xx purge"

### DISCUSSION THREAD

**Discussion by blade83lee** [databendlabs/databend#14030] [Q&A]
i use golang (   "insert into xx" )  send data to databend  continuously,   
crontab 10 minutes  "optimize table xx compact"
now i want to  optimize table xx purge， it take long time
![企业微信截图_17026121634363](https://github.com/datafuselabs/databend/assets/88416508/1ea46753-9a22-498f-9a8b-c08f5a72d33c)
3 nodes databend-query 8C 16G

**Comment by blade83lee** [comment]
i have two large table
![企业微信截图_17026128167068](https://github.com/datafuselabs/databend/assets/88416508/5454e137-6208-49ed-9251-452d5dfa9ef9)

**Comment by blade83lee** [comment]
![企业微信截图_17026128427748](https://github.com/datafuselabs/databend/assets/88416508/06c8c4d2-e960-4eec-82e9-41a55f846e9f)

**Comment by blade83lee** [comment]
bug s3 have
![企业微信截图_17026128999323](https://github.com/datafuselabs/databend/assets/88416508/89fc9964-f4e6-4aaa-82e9-f1864818007f)
so， why

**Comment by bohutang** [comment]
It's better to add limit to purge and run many times: `OPTIMIZE TABLE t PURGE LIMIT 100`
